enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,416 @@
+}
+
+type NodeDeviceMap List struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []NodeDeviceMap `json:"items"`
+}
+
+type NodeDeviceMap struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              NodeDeviceMapSpec   `json:"spec"`
+	Status            NodeDeviceMapStatus `json:"status,omitempty"`
+}
+
+type NodeDeviceMapSpec struct {
+	NodeName string `json:"nodeName"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type NodeDeviceMapStatus struct {
+	Phase string `json:"phase,omitempty"`
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name       string   `json:"name,omitempty"`
+	Model      string   `json:"model,omitempty"`
+	Vendor     string   `json:"vendor,omitempty"`
+	Kname      string   `json:"kname,omitempty"`
+	Pkname     string   `json:"pkname,omitempty"`
+	Type       string   `json:"type,omitempty"`
+	UUID       string   `json:"uuid,omitempty"`
+	Tran       string   `json:"tran,omitempty"`
+	Ro         string   `json:"ro,omitempty"`
+	Parttype   string   `json:"parttype,omitempty"`
+	Serial     string   `json:"serial,omitempty"`
+	Size       string   `json:"size,omitempty"`
+	Mountpoint string   `json:"mountpoint,omitempty"`
+	Rm         string   `json:"rm,omitempty"`
+	Children   []Device `json:children,omitempty`
+}
+```
+
+#### Workflow of Device Discovery
+- The discovery of local devices will be done by the `LocalVolumeDiscovery` CR. Once the `LocalVolumeDiscovery` CR is deployed, the controller of the `LocalVolumeDiscovery` CR will be managing a `NodeDeviceMap` CR for each node that matched the `nodeSelector`. The `NodeDeviceMap` CR will be exposing the available local devices of the node in its status. These information will mostly be the output of  `lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint`
+
+Example of `LocalVolumeDiscovery` CR:
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeDiscovery
+metadata:
+  name: example-autodetect
+  namespace: localStorage
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+status:
+  phase: Discovered
+```
+
+Example of `NodeDeviceMap` CR after detection:
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: NodeDeviceMap
+metadata:
+  name: example-volume-detect-1
+  namespace: localStorage
+spec:
+  nodeName: worker-0
+  tolerations: null
+Status:
+  Phase: Discovered
```

Comment:
it will be a one time job

Reply:
Can you please explicitly emphasize it somewhere? @jarrpa, FYI, this is a good place to argue otherwise. How is GUI to refresh it? How often?

Reply:
its already mentioned here https://github.com/openshift/enhancements/pull/237/files#diff-ea6b51c2a2234cd3fb1280d7332fd70aR83

Reply:
> Can you please explicitly emphasize it somewhere? @jarrpa, FYI, this is a good place to argue otherwise. How is GUI going to refresh it? How often?

@jsafrane Based on the latest discussion we had: 
- running device discovery continuously will be expensive. It should be a one time thing. Don't allow this operation to be performed multiple times even from the UI. 
- If the user has selected the wrong nodes or wants to change the selected nodes, then they can cancel out of the operation. In this case the device Discovery CRs will be deleted and we can start the operation again with the new device discovery CR. 
@jarrpa  Can you please confirm this behavior again.